src/v3/src/util/theme.ts

```diff
@@ -10,170 +10,126 @@
-      set(theme, 'palette.primary.lighter', customTokens.PalettePrimaryLighter ?? p.lighter);
-      set(theme, 'palette.primary.light', customTokens.PalettePrimaryLight ?? p.light);
-      set(theme, 'palette.primary.main', customTokens.PalettePrimaryMain ?? p.main);
-      set(theme, 'palette.primary.dark', customTokens.PalettePrimaryDark ?? p.dark);
-      set(theme, 'palette.primary.contrastText', p.contrastText);
-    }
-    if (customTokens.PaletteDangerMain) {
-      const p = generatePalette(customTokens.PaletteDangerMain);
-      set(theme, 'palette.error.lighter', customTokens.PaletteDangerLighter ?? p.lighter);
-      set(theme, 'palette.error.light', customTokens.PaletteDangerLight ?? p.light);
-      set(theme, 'palette.error.main', customTokens.PaletteDangerMain ?? p.main);
-      set(theme, 'palette.error.dark', customTokens.PaletteDangerDark ?? p.dark);
-      set(theme, 'palette.error.contrastText', p.contrastText);
-    }
-    if (customTokens.PaletteWarningMain) {
-      const p = generatePalette(customTokens.PaletteWarningMain);
-      set(theme, 'palette.warning.lighter', customTokens.PaletteWarningLighter ?? p.lighter);
-      set(theme, 'palette.warning.light', customTokens.PaletteWarningLight ?? p.light);
-      set(theme, 'palette.warning.main', customTokens.PaletteWarningMain ?? p.main);
-      set(theme, 'palette.warning.dark', customTokens.PaletteWarningDark ?? p.dark);
-      set(theme, 'palette.warning.contrastText', p.contrastText);
-    }
-    if (customTokens.PaletteSuccessMain) {
-      const p = generatePalette(customTokens.PaletteSuccessMain);
-      set(theme, 'palette.success.lighter', customTokens.PaletteSuccessLighter ?? p.lighter);
-      set(theme, 'palette.success.light', customTokens.PaletteSuccessLight ?? p.light);
-      set(theme, 'palette.success.main', customTokens.PaletteSuccessMain ?? p.main);
-      set(theme, 'palette.success.dark', customTokens.PaletteSuccessDark ?? p.dark);
-      set(theme, 'palette.success.contrastText', p.contrastText);
-    }
+    tokensOverride.PalettePrimaryLighter ??= p?.CustomHue50;
+    tokensOverride.PalettePrimaryHighlight ??= p?.CustomHue100;
+    tokensOverride.PalettePrimaryLight ??= p?.CustomHue300;
+    tokensOverride.PalettePrimaryMain ??= p?.CustomHue500;
+    tokensOverride.FocusOutlineColorPrimary ??= p?.CustomHue500;
+    tokensOverride.BorderColorPrimaryControl ??= p?.CustomHue500;
+    tokensOverride.TypographyColorAction ??= p?.CustomHue600;
+    tokensOverride.PalettePrimaryText ??= p?.CustomHue600;
+    tokensOverride.BorderColorPrimaryDark ??= p?.CustomHue700;
+    tokensOverride.PalettePrimaryDark ??= p?.CustomHue700;
+    tokensOverride.PalettePrimaryDarker ??= p?.CustomHue800;
+    tokensOverride.PalettePrimaryHeading ??= p?.CustomHue900;
   }
 
+  // Odyssey 1.x does not export their theme, but we can recreate it
+  const baseOdysseyTheme = createOdysseyMuiTheme({
+    odysseyTokens: { ...Tokens, ...tokensOverride },
+  });
+
```

Comment:
One niche thing to note: palette generation is now limited to the design tokens layer and will not apply at the base MUI theming layer (the base MUI theme appears in code via `theme.palette.*`). This is fine because even though Odyssey does create a MUI theming layer via this `createOdysseyMuiTheme` call, they use their design tokens to achieve essentially all of their styling. 

TLDR: If the customer begins seeing any styling that can be traced to the base MUI theme layer, there's probably something wrong and this theming is just a fallback